src/completeComment.ts

```diff
@@ -56,35 +56,17 @@ export const convertCommentToTxtNode = (
 };
 
 export const isAppearedBeforeNode = (
-  nodeLocation: TxtNodeLineLocation,
-  commentLocation: TxtNodeLineLocation
+  nodeRange: TextNodeRange,
+  commentRange: TextNodeRange
```

Comment:
What is it difference between `TxtNodeLineLocation` and `TextNodeRange`?
Please tell me why this change solved the issue? Thanks.

Reply:
While debugging, I found that when parsing the sample in the issue, the comment was inserted in an unexpected place. This seemed to have introduced a node with an invalid location into the AST.

<details><summary>Generated AST in master branch (Inserted comment has invalid location)</summary>

```js
      {
        type: 'Document',
        raw: 'A%B\nC',
        range: [ 0, 5 ],
        loc: { start: { line: 1, column: 0 }, end: { line: 2, column: 0 } },
        children: [
          {
            loc: { start: { line: 1, column: 0 }, end: { line: 2, column: 1 } },
            range: [ 0, 5 ],
            raw: 'A%B\nC',
            type: 'Paragraph',
            children: [
              {
                loc: {
                  start: { line: 1, column: 0 },
                  end: { line: 1, column: 1 }
                },
                range: [ 0, 1 ],
                raw: 'A',
                type: 'Str',
                value: 'A'
              },
              {
                loc: {
                  start: { line: 1, column: 1 },
                  end: { line: 2, column: 0 }
                },
                range: [ 1, 4 ],
                raw: '%B\n',
                type: 'Html',
                value: '%B\n'
              },
              {
                loc: {
                  start: { line: 2, column: 0 },
                  end: { line: 2, column: 1 }
                },
                range: [ 4, 5 ],
                raw: 'C',
                type: 'Str',
                value: 'C'
              }
            ]
          },
          {
            loc: { start: { line: 2, column: 1 }, end: { line: 1, column: 1 } },
            range: [ 5, 1 ],
            raw: '',
            type: 'Html',
            value: ''
          },
          {
            loc: { start: { line: 1, column: 1 }, end: { line: 2, column: 0 } },
            range: [ 1, 4 ],
            raw: '%B\n',
            value: 'B',
            type: 'Comment'
          }
        ]
      }
```

</details>

I found that this might be caused by the wrong algorithm of `insertComment`. When I thought about it, I realized that the algorithm would be simpler if I used range rather than location. After this change, the generated AST becomes valid.

<details><summary>Generated AST in this branch</summary>

```js
      {
        type: 'Document',
        raw: 'A%B\nC',
        range: [ 0, 5 ],
        loc: { start: { line: 1, column: 0 }, end: { line: 2, column: 0 } },
        children: [
          {
            loc: { start: { line: 1, column: 0 }, end: { line: 2, column: 1 } },
            range: [ 0, 5 ],
            raw: 'A%B\nC',
            type: 'Paragraph',
            children: [
              {
                loc: {
                  start: { line: 1, column: 0 },
                  end: { line: 1, column: 1 }
                },
                range: [ 0, 1 ],
                raw: 'A',
                type: 'Str',
                value: 'A'
              },
              {
                loc: {
                  start: { line: 1, column: 1 },
                  end: { line: 2, column: 0 }
                },
                range: [ 1, 4 ],
                raw: '%B\n',
                value: 'B',
                type: 'Comment'
              },
              {
                loc: {
                  start: { line: 2, column: 0 },
                  end: { line: 2, column: 1 }
                },
                range: [ 4, 5 ],
                raw: 'C',
                type: 'Str',
                value: 'C'
              }
            ]
          }
        ]
      }
```

</details>

I use `npm link` and test sample TeX source. textlint does not show any error.

Reply:
Hmm, you mean that this pr changes the algorithm to be work expectedly with TextNodeRange, right? If so, I need to review more carefully. If it is just minor (or trivial) changes, I can approve immediately.

Reply:
It seems that the previous implementation was complicated and contained bugs. I consider this to be just a bug fix, not a major algorithm change :) (In fact, the unit tests from before are working fine without any changes.)